### PR TITLE
Closes #1747 Update: Following deprecation process for multiple funct…

### LINKED ISSUE #1747: General Issue: Review Programs, vignettes and templates for deprecation functions
### Background Information

Go through the programs, templates, vignettes and tests to ensure that all of the functions and/or parameters that were deprecated are following the deprecation process outlined in the programming strategy (screenshot below). (related functions and/or parameters will be added to this issue to keep a running list)
image

### Definition of Done

- [x]  All programs, templates, vignettes and tests with deprecation functions are appropriately updated
- [x] All functions and/or parameters with **deprecated_warn()** from previous release have been updated to **deprecated_stop()** no longer in any vignettes or templates
- [x] Following functions need to removed for release Q2-2023 admiral version 0.11.0 because they were deprecated in version 0.10.0 with phase 2:
`derive_vars_merged_dt()`
`derive_vars_merged_dtm()`
`derive_var_agegr_ema()`
`derive_var_agegr_fda()`
`derive_param_first_event()`
`derive_derived_param()`

### PR DESCRIPTION
**Pull request by sadchla-codes**
…ions

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
I think we want the deprecations tests to be moved into the test files and remove the `deprecation.R` file.  @bundfussr is that correct?

**Comment by bms63** [comment]
@sadchla-codes if only 1 test left, then lets move to correct file and remove this deprecation file.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: NEWS.md [hunk 1]
```diff
@@ -8,8 +8,27 @@
...
+
+  - `derive_var_disposition_status()`
+  - `derive_vars_disposition_reason()`
+  - `format_eoxxstt_default()`
+  - `format_reason_default()`
+  - `derive_var_worst_flag()`
+
 ## Various
 
+`vars()` which was used in the admiral function calls that expected a list of quosures have being removed (#1747).
```
Comment (zdz2101) [inline NEWS.md:31]:
```suggestion
`vars()` which was used in the admiral function calls that expected a list of quosures has being removed (#1747).
```

File: NEWS.md [hunk 2]
```diff
@@ -8,8 +8,27 @@
...
 
+- The following functions, which were deprecated in previous `{admiral}` versions, have been removed (#1747):
+
+  - `derive_vars_merged_dt()`
+  - `derive_vars_merged_dtm()`
+  - `derive_var_agegr_ema()`
+  - `derive_var_agegr_fda()`
+  - `derive_param_first_event()`
+  - `derive_derived_param()`
+  - `derive_var_confirmation_flag()`
```
Comment (zdz2101) [inline NEWS.md:19]:
```suggestion
  - `derive_var_confirmation_flag()`
  - `filter_confirmation()`
```

Should this be included?

**Review by zdz2101 — CHANGES_REQUESTED** [review]
Small stuff, otherwise looks good

**Comment by bms63** [comment]
@sadchla-codes please merge - two approvals is good for me!!

**Pull request merged**